common/Unit.hpp

```diff
@@ -54,6 +54,83 @@ namespace Poco
 class Session;
 class StorageBase;
 
+/// AtomicSharedPtr provides atomic operation for `std::shared_ptr<T>` similar to `std::atomic<std::shared_ptr<T>>`
+/// It further provides thread-safe `assignIfNull` and `assignIfNotNull` atomic conditional assignment methods.
+/// Implementation uses `std::mutex` to synchronize `std::shared_ptr<T>` access.
+template <class T>
+class AtomicSharedPtr
```

Comment:
[`std::atomic<std::shared_ptr>`](https://en.cppreference.com/w/cpp/memory/shared_ptr/atomic2)?

I'm sure this was fun to write, but the standard has this already... assuming we need it at all. I'm not convinced we are fixing the right code.

Reply:
Both, I already elaborated on atomic<shared_ptr> and the overall reasoning.
This AtomicSharedPtr was just offered to sooth acceptance, encapsulating the mutex avoiding abuse
_and_ to offer an atomic easy to use `assingIfNull` method.

Added more context to the conversation in the original branch.
Closing this without merging.